src/promptflow-tracing/promptflow/tracing/_retrieval.py

```diff
@@ -0,0 +1,56 @@
+import json
+from typing import Callable
+
+from opentelemetry.trace import Span
+
+from ._span_enricher import SpanEnricher, SpanEnricherManager
+from ._trace import TraceType, _traced
+
+keys = ["id", "score", "content", "metadata"]
+
+
+class RetrievalSpanEnricher(SpanEnricher):
```

Comment:
Consider separate into two PRs?
* One is for adding SpanEnricher and refactoring the existing spans.
* Another one is adding RetrievalSpan.

Reply:
Refactored.